### PR TITLE
Removed all support for DataSet, DataTable, DataRow and such

### PR DESCRIPTION
**Pull request by dennisdoomen**
This was moved to a new repository and package https://github.com/fluentassertions/fluentassertions.datasets

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by IT-VBFK** [comment]
On the road to v7? 🎉

**Comment by jnyrup** [comment]
> On the road to v7? 🎉

We will soon release 6.12 and then start working on v7

File: Tests/FluentAssertions.Specs/FluentAssertions.Specs.csproj [hunk 1]
```diff
@@ -75,4 +75,5 @@
     <ProjectReference Include="..\AssemblyB\AssemblyB.csproj" />
   </ItemGroup>
 
+
```
Comment (lg2de) [inline Tests/FluentAssertions.Specs/FluentAssertions.Specs.csproj:78]:
```suggestion
```

File: docs/_pages/releases.md [hunk 2]
```diff
@@ -17,6 +15,8 @@ sidebar:
 
 ### Breaking Changes (for users)
 
+* Moved support for `DataSet`, `DataTable`, `DataRow` and `DataColum` into a new package `FluentAssertions.DataSet` - [#1111](https://github.com/fluentassertions/fluentassertions/pull/2111)
```
Comment (jnyrup) [inline docs/_pages/releases.md:18]:
```suggestion
* Moved support for `DataSet`, `DataTable`, `DataRow` and `DataColumn` into a new package `FluentAssertions.DataSet` - [#2267(https://github.com/fluentassertions/fluentassertions/pull/2267)
```

File: docs/_pages/releases.md [hunk 3]
```diff
@@ -7,8 +7,6 @@ sidebar:
   nav: "sidebar"
 ---
 
-## 7.0 Alpha 1
-
```
Comment (jnyrup) [inline docs/_pages/releases.md:11]: keep?

File: Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt [hunk 4]
```diff
@@ -1062,8 +1062,6 @@ namespace FluentAssertions.Equivalency
         public TSelf IncludingInternalProperties() { }
         public TSelf IncludingNestedObjects() { }
         public TSelf IncludingProperties() { }
-        public TSelf RespectingDeclaredTypes() { }
-        public TSelf RespectingRuntimeTypes() { }
```
Comment (jnyrup) [inline Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt:1066]: rerun ApprovalTests

**Review by jnyrup — COMMENTED** [review]
A few more things to remove:

* `ConstraintCollectionEquivalencyStep`
* `ConstraintEquivalencyStep`
* 3 references to `System.Data.DataSetExtensions`

File: docs/_pages/releases.md [hunk 5]
```diff
@@ -17,6 +17,8 @@ sidebar:
 
 ### Breaking Changes (for users)
 
+* Moved support for `DataSet`, `DataTable`, `DataRow` and `DataColum` into a new package `FluentAssertions.DataSet` - [#2267](https://github.com/fluentassertions/fluentassertions/pull/2267)
```
Comment (jnyrup) [inline docs/_pages/releases.md:20]:
```suggestion
* Moved support for `DataSet`, `DataTable`, `DataRow` and `DataColumn` into a new package `FluentAssertions.DataSet` - [#2267](https://github.com/fluentassertions/fluentassertions/pull/2267)
```

**Review by jnyrup — APPROVED** [review]
A single typo plus Lukas' suggestion.

**Pull request merged**